info/v2/container.go

```diff
@@ -86,8 +86,11 @@ type ContainerStats struct {
 	// The time of this stat point.
 	Timestamp time.Time `json:"timestamp"`
 	// CPU statistics
-	HasCpu bool        `json:"has_cpu"`
-	Cpu    v1.CpuStats `json:"cpu,omitempty"`
+	HasCpu bool `json:"has_cpu"`
+	// In nanoseconds (aggregated)
+	Cpu v1.CpuStats `json:"cpu,omitempty"`
+	// In nanocores per second (instantaneous)
+	CpuInst v1.CpuStats `json:"cpu_inst,omitempty"`
```

Comment:
Let's do it in v2. Kubelet can switch to using v2. We should consider doing
this for other instantaneous metrics as well.

On Fri, Aug 21, 2015 at 11:58 AM, Daniel Martí <notifications@github.com>
wrote:

> In info/v2/container.go
> <https://github.com/google/cadvisor/pull/861#discussion_r37666630>:
>
> > @@ -86,8 +86,11 @@ type ContainerStats struct {
> >  	// The time of this stat point.
> >  	Timestamp time.Time `json:"timestamp"`
> >  	// CPU statistics
> > -	HasCpu bool        `json:"has_cpu"`
> > -	Cpu    v1.CpuStats `json:"cpu,omitempty"`
> > +	HasCpu bool `json:"has_cpu"`
> > +	// In nanoseconds (aggregated)
> > +	Cpu v1.CpuStats `json:"cpu,omitempty"`
> > +	// In nanocores per second (instantaneous)
> > +	CpuInst v1.CpuStats `json:"cpu_inst,omitempty"`
>
> I would have assumed that v1 was frozen by now. Otherwise, shouldn't
> changes in v1 be in v2 too?
>
> Would moving kubelet to v2 be an issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/861/files#r37666630>.
>